container.go

```diff
@@ -747,11 +747,17 @@ func (c *Client) ExportContainer(opts ExportContainerOptions) error {
 
 // NoSuchContainer is the error returned when a given container does not exist.
 type NoSuchContainer struct {
-	ID string
+	ID  string
+	Err error
 }
 
 func (err *NoSuchContainer) Error() string {
-	return "No such container: " + err.ID
+	switch err.Err.(type) {
+	case error:
+		return "No such container: " + err.ID + " - " + err.Err.Error()
+	default:
+		return "No such container: " + err.ID
+	}
```

Comment:
Yeah, looks silly now. Look how they did it: https://github.com/docker/docker/blob/master/api/client/utils.go#L112
They have two types of generic errors, one if body present, another if body is empty.
We can do the same check, so if body is empty, we give `NoSuchContainer`, if there is some data in body, give some generic error, putting body there.